File: BookShop.BusinessLogic/Models/PrintedEditionModels/PrintedEditionRequestModel.cs
﻿using BookShop.DataAccess.Models.RequestParameters;
using static BookShop.DataAccess.Common.Enums.EntityFields;

namespace BookShop.BusinessLogic.PrintedEditionModels
{
    public class PrintedEditionRequestModel
    {
        public int Page { get; set; }
        public int PageSize { get; set; }
        public string KeyWord { get; set; }
        public decimal PriceTo { get; set; }
        public decimal PriceFrom { get; set; }
        public SortCriteria SortCriteria { get; set; }
        public PrintedEditionType PrintedEditionType { get; set; }

        internal PrintedEditionRequestParameters MapToRequestParameters()
        {
            PrintedEditionRequestParameters parameters = new PrintedEditionRequestParameters();
            if (Page != default(int))
            {
                parameters.Page = Page;
            }
            if (PageSize != default(int))
            {
                parameters.PageSize = PageSize;
            }
            if (!string.IsNullOrWhiteSpace(KeyWord))
            {
                parameters.KeyWord = KeyWord;
            }
            if (PriceFrom != default(int))
            {
                parameters.PriceFrom = PriceFrom;
            }
            if (PriceTo != default(int))
            {
                parameters.PriceTo = PriceTo;
            }
            if (SortCriteria != SortCriteria.None)
            {
                parameters.SortCriteria = SortCriteria;
            }
            if (PrintedEditionType != PrintedEditionType.None)
            {
                parameters.PrintedEditionType = PrintedEditionType;
            }
            return parameters;
        }
    }
}

File: BookShop.Presentation/wwwroot/ClientApp/src/app/models/printedEditionModels/printedEditionRequestModel.ts
import { SortCriteria } from 'src/app/enums/sortCriteria';
import { PrintedEditionType } from 'src/app/enums/printedEditionType';
import { PrintedEditionModel } from 'src/app/models/printedEditionModels/printedEditionModel';

export class PrintedEditionRequestModel {
    public page: number;
    public pageSize: number;
    public keyWord: string;
    public priceTo: number;
    public priceFrom: number;
    public withPagination: boolean;
    public sortCriteria: SortCriteria;
    public printedEditionType: PrintedEditionType;
    public printedEditionIgnoreList: PrintedEditionModel[];

    constructor() {
        this.initializeDefaultModel();
    }

    private initializeDefaultModel() {
        this.page = 1;
        this.pageSize = 6;
        this.keyWord = '';
        this.priceTo = 0;
        this.priceFrom = 0;
        this.withPagination = true;
        this.sortCriteria = SortCriteria.None;
        this.printedEditionType = PrintedEditionType.None;
        this.printedEditionIgnoreList = [];
    }
}
